File: app/src/main/java/com/example/imhere/PreviewActivity.java
package com.example.imhere;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.AsyncTask;
import android.os.Bundle;

import java.util.concurrent.TimeUnit;

public class PreviewActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_preview);

        ScreenDelayTask screenDelayTask = new ScreenDelayTask();
        screenDelayTask.execute();
    }

    public class ScreenDelayTask extends AsyncTask<Void, Void, Void> {

        @Override
        protected Void doInBackground(Void... voids) {
            try {
                TimeUnit.SECONDS.sleep(1);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
            return null;
        }

        @Override
        protected void onPostExecute(Void result){
            SharedPreferences sp = getSharedPreferences("authentication", MODE_PRIVATE);
            if (!sp.contains("authentication") || !sp.getBoolean("authentication", true)){
                startActivity(new Intent(PreviewActivity.this, LoginActivity.class));
                PreviewActivity.super.finish();
            }
            else {
                SharedPreferences.Editor ed = sp.edit();
                ed.putBoolean("button_lock", true);

                PreviewActivity.super.finish();
            }
        }
    }
}

File: app/src/main/java/com/example/imhere/PreviewActivity.kt
package com.example.imhere

import android.content.ContentValues
import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.coroutines.*

class PreviewActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_preview)
    }

    override fun onResume() {
        super.onResume()
        GlobalScope.launch {
            delay(1000)
            withContext(Dispatchers.Main){
                val sp = getSharedPreferences("authentication", Context.MODE_PRIVATE)
                if (sp.contains("authentication") && sp.getBoolean("authentication", false)) {
                    if (sp.getInt("status", 2) == 0) startActivity(Intent(this@PreviewActivity, StudentActivity::class.java))
                    else startActivity(Intent(this@PreviewActivity, AddInterviewActivity::class.java))
                } else startActivity(Intent(this@PreviewActivity, LoginActivity::class.java))
                super@PreviewActivity.finish()
            }
        }
    }
}